internal/buffer/settings.go

```diff
@@ -50,8 +52,16 @@ func (b *Buffer) DoSetOptionNative(option string, nativeValue interface{}) {
 		return
 	}
 
+	oldValue := b.Settings[option]
```

Comment:
This would be useful for the above `reflect.DeepEqual` check as well.

Reply:
But, the old value was already checked.
Maybe I don't get, what you're trying to tell.

Reply:
I meant, we could move this line above the `reflect.DeepEqual` check, and then we could use `oldValue` in the `reflect.DeepEqual` check as well, instead of doing lookup of `b.Settings[option]` twice.

Moreover, I've just realized: since we've indeed already checked the equality above (and even in a more accurate way), it means that this `oldValue != nativeValue` check is not needed at all?

Reply:
> instead of doing lookup of `b.Settings[option]` twice.

Ah yeah, got your point...since we realized in different discussions that this check is very expensive.

> it means that this `oldValue != nativeValue` check is not needed at all?

Indeed, this wasn't present in the moment the PR was created.